connection.go

```diff
@@ -78,11 +95,11 @@ func (mc *mysqlConn) Begin() (driver.Tx, error) {
 
 func (mc *mysqlConn) Close() (err error) {
 	// Makes Close idempotent
-	if mc.netConn != nil {
+	if !mc.isBroken() {
 		err = mc.writeCommandPacket(comQuit)
 	}
 
-	mc.cleanup()
+	mc.cleanup(errors.New("mysql: connection is closed"))
```

Comment:
Might a user see this error?
If yes, please define it in `errors.go`

If it is completely internal, please define it at the top of the file.